### ISSUE TITLE
"HAS NO VALUE" condition is not working with prometheus query in Grafana

### ISSUE THREAD

**Issue by mohddanish211** [grafana/grafana#45786]
Hi,  I am using Grafana to create alerts for different scenarios. I am using the Prometheus as data source and wrote a Prometheus query which will return the data volume of particular system. I want to create alert if the system is not sending any data so I have used the condition "HAS NO VALUE" for the alert. Since the data volume varies for different systems, I cant put a certain number in the BELOW condition and thus used the "HAS NO VALUE" condition, But when I test my alert condition it come always false.

Query used:
sum(increase(total{job="prod", system="abc", message="receive"}[1h])) by (system)

![image](https://user-images.githubusercontent.com/16339561/155332188-24e5de11-5802-4732-9011-c91fa7f72a71.png)

![image](https://user-images.githubusercontent.com/16339561/155332445-3276070c-8ce6-46e2-a554-ea36ff0b0a64.png)

- Grafana version: 8.1.0-beta3
- Data source type & version: Prometheus
- OS Grafana is installed on: linux
- User OS & Browser: chrome

**Comment by zuchka** [comment]
@mohddanish211 are you still encountering this error?

Please list ALL configuration options related to alerting. You can find these in the `Alerting` and `Unified Alerting` sections of Grafana's config file. If you are now using or have previously used the beta version of `ngalert` (released with Grafana 8), please note that too.
  - you can use [this table](https://github.com/grafana/grafana/pull/42200#issue-1062269188) to better understand how configuration options can interact with each other

**Comment by mohddanish211** [comment]
@zuchka Thanks for the response. I have checked the table that you have mentioned and was able to achieve the results. 

I have one more query regarding the Prometheus. I am looking for a query/function in Prometheus that will give the relative change in the data. I have checked the Prometheus documentation but not able to find the suitable solution. I would like to create an alert based on the relative change not with the fixed value(for example if the messages for system A is less than 50 then send me an alert). Let me explain with the example here:

![image](https://user-images.githubusercontent.com/16339561/159669074-4585367a-be09-434e-b676-429d61caf448.png)


The drop of volume for system A is more concerning as it more drastic on a relative scale while both systems have same message volume around 10:00 hr.  Therefore, I need to develop a Prometheus query to track the relative change of messages volume per interval. Any pointers will be really helpful. Thanks in advance.
